server/protocols/jmap/src/main/java/org/apache/james/jmap/model/CreationMessage.java

```diff
@@ -311,18 +311,32 @@ public ZonedDateTime getDate() {
         return attachedMessages;
     }
 
-    public boolean isValid() {
-        return validate().isEmpty();
+    public boolean isValid(ImmutableList<String> allowedFromMailAddresses) {
+        return validate(allowedFromMailAddresses).isEmpty();
     }
 
-    public List<ValidationResult> validate() {
+    public List<ValidationResult> validate(ImmutableList<String> allowedFromMailAddresses) {
         ImmutableList.Builder<ValidationResult> errors = ImmutableList.builder();
         assertValidFromProvided(errors);
         assertAtLeastOneValidRecipient(errors);
         assertSubjectProvided(errors);
+        assertUsedAllowedFromMailAddress(errors, allowedFromMailAddresses);
         return errors.build();
     }
 
+    private void assertUsedAllowedFromMailAddress(ImmutableList.Builder<ValidationResult> errors, ImmutableList<String> allowedFromMailAddresses) {
```

Comment:
It's not a validity concern, it's an access control one. It doesn't belong to validate method.